robot/client/client_test.go

```diff
@@ -98,6 +98,117 @@ var finalResources = []resource.Name{
 
 var pose1 = spatialmath.NewZeroPose()
 
+type mockRPCSubtypesUnimplemented struct {
+	pb.UnimplementedRobotServiceServer
+}
+
+func (ms *mockRPCSubtypesUnimplemented) ResourceNames(
+	ctx context.Context, _ *pb.ResourceNamesRequest) (*pb.ResourceNamesResponse, error) {
+	board := board.Named("micro-rdk")
+	rNames := []*commonpb.ResourceName{
+		{
+			Namespace: string(board.API.Type.Namespace),
+			Type:      board.API.Type.Name,
+			Subtype:   board.API.SubtypeName,
+			Name:      board.Name,
+		},
+	}
+	return &pb.ResourceNamesResponse{Resources: rNames}, nil
+}
+
+type mockRPCSubtypesImplemented struct {
+	mockRPCSubtypesUnimplemented
+}
+
+func (ms *mockRPCSubtypesImplemented) ResourceRPCSubtypes(
+	ctx context.Context, _ *pb.ResourceRPCSubtypesRequest,
+) (*pb.ResourceRPCSubtypesResponse, error) {
+	return &pb.ResourceRPCSubtypesResponse{}, nil
+}
+
+func TestUnimplementedRPCSubtypes(t *testing.T) {
+	var client1 *RobotClient // test implemented
+	var client2 *RobotClient // test unimplemented
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second*1)
+	defer cancel()
+
+	var port1 int
+	var port2 int
+	var err error
+	logger := logging.NewTestLogger(t)
+	for i := 0; i < 10; i++ {
+		port1, err = utils.TryReserveRandomPort()
```

Comment:
Unfortunately `err` here will not be `non-nil` if you collided with another test. I think you'll want to move the `net.Listen` and `rpc.NewServer` calls into the `for` loops, and use a utility like [WaitForServing](https://github.com/viamrobotics/rdk/blob/main/testutils/robottestutils/robot_utils.go#L98). I filed RSDK-6749 to improve that `WaitForServing` utility so we hopefully won't have to do this in the future.